parser.go

```diff
@@ -422,19 +429,86 @@ func (r *Rule) network(key item, l *lexer) error {
 	}
 	switch key.typ {
 	case itemSourceAddress:
-		r.Source.Nets = append(r.Source.Nets, items...)
+		if iPAddressesValid(items[:]...) {
+			r.Source.Nets = append(r.Source.Nets, items...)
```

Comment:
if you make the below changes I believe you can just `if ipAddressesValid(items) { ...`